lewis_emulators/aldn1000/interfaces/stream_interface.py

```diff
@@ -0,0 +1,166 @@
+        }
+
+    @if_input_error
+    def basic_get_response(self, address, data=None, units=None):
+        # The device requires very specific formatting(0000. - 1000.) for floats, so we restrict the float output to
+        # to the first 5 characters of the float converted to a string. Hackish, but the only way we found to
+        # quickly implement this without writing our own formatter.
+        if units is not None:
+            return STX + '{address:02d}{status}{data}{units}'.format(address=address, status=self.device.status,
+                                                                     data=data, units=units) + ETX
+        else:
+            return STX + '{address:02d}{status}{data}'.format(address=address, status=self.device.status,
+                                                              data=data) + ETX
+
+    @if_input_error
+    def basic_set_response(self, address):
+        response = STX + '{:02d}{status}'.format(address, status=self.device.status) + ETX
+        return response
+
+    @if_connected
+    def format_data(self, float):
+        # Restrict a float to only the first 5 characters
+        return '{formatted_value:.5s}'.format(formatted_value=str(float))
+
+    @if_connected
+    def set_pump(self, address, action):
+        self.device.address = address
+        if action == 'STP':
+            self.device.pump = 'STP'
+            if self.device.status == 'P':  # Currently paused
+                self.device.status = 'S'  # Stop
+            elif self.device.status == 'S':
+                pass
+            else:
+                self.device.status = 'P'  # Pause
+        elif action == 'RUN':
+            self.device.pump = 'RUN'
+            if self.device.direction == 'Infusing':
+                self.device.status = 'I'
+            else:
+                self.device.status = 'W'
+        else:
+            print('An error occurred while trying to start/stop the pump')
+        return self.basic_set_response(address)
+
+    @if_connected
+    def get_status(self, address):
+        return self.basic_get_response(address, self.device.status)
+
+    @if_connected
+    def set_pump_on(self, address):
+        self.device.address = address
+        self.device.pump = 'ON'
+
+    @if_connected
+    def set_pump_off(self, address):
+        self.device.address = address
+        self.device.pump = 'OFF'
+
+    @if_connected
+    def get_diameter(self, address):
+        return self.basic_get_response(address, self.format_data(self.device.diameter))
+
+    @if_connected
+    def set_diameter(self, address, diameter):
+        self.device.address = address
+        self.device.diameter = diameter
+        if diameter > 14.00:
+            self.device.volume_units = 'mL'
+        else:
+            self.device.volume_units = 'uL'
+        return self.basic_set_response(address)
+
+    @if_connected
+    def get_volume(self, address):
+        if self.device.volume_units == 'mL':
```

Comment:
Why are you storing this as lowercase on the device? If you don't we can just respond with the volume without doing this check?